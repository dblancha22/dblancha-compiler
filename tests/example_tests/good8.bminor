//test arrays
f:array [5]integer = { 1, 2, 3, 4, 5};
f:array[5]char = {'a', 'b', 'c', 'd', 'e'};
f:array[5] boolean = {false, false, true, true, true};
f:array [] 	string = {"this", "i", "s", "a,", "test", "."};
