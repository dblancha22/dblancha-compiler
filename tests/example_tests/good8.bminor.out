COMMENT //test arrays
IDENTIFIER f
COLON :
ARRAY array
OPEN_SQUARE [
INTEGER_LITERAL 5
CLOSE_SQUARE ]
INTEGER integer
EQUALS =
OPEN_CURLY {
INTEGER_LITERAL 1
NOT ,
INTEGER_LITERAL 2
NOT ,
INTEGER_LITERAL 3
NOT ,
INTEGER_LITERAL 4
NOT ,
INTEGER_LITERAL 5
CLOSE_CURLY }
SEMICOLON ;
IDENTIFIER f
COLON :
ARRAY array
OPEN_SQUARE [
INTEGER_LITERAL 5
CLOSE_SQUARE ]
CHAR char
EQUALS =
OPEN_CURLY {
CHAR_LITERAL 'a'
NOT ,
CHAR_LITERAL 'b'
NOT ,
CHAR_LITERAL 'c'
NOT ,
CHAR_LITERAL 'd'
NOT ,
CHAR_LITERAL 'e'
CLOSE_CURLY }
SEMICOLON ;
IDENTIFIER f
COLON :
ARRAY array
OPEN_SQUARE [
INTEGER_LITERAL 5
CLOSE_SQUARE ]
BOOLEAN boolean
EQUALS =
OPEN_CURLY {
FALSE false
NOT ,
FALSE false
NOT ,
TRUE true
NOT ,
TRUE true
NOT ,
TRUE true
CLOSE_CURLY }
SEMICOLON ;
IDENTIFIER f
COLON :
ARRAY array
OPEN_SQUARE [
CLOSE_SQUARE ]
STRING string
EQUALS =
OPEN_CURLY {
STRING_LITERAL this
NOT ,
STRING_LITERAL i
NOT ,
STRING_LITERAL s
NOT ,
STRING_LITERAL a,
NOT ,
STRING_LITERAL test
NOT ,
STRING_LITERAL .
CLOSE_CURLY }
SEMICOLON ;
